R/implicit_assignment_linter.R

```diff
@@ -75,6 +92,14 @@ implicit_assignment_linter <- function(except = c("bquote", "expression", "expr"
   if (allow_lazy) {
     xpath <- paste0(xpath, "[not(ancestor::expr/preceding-sibling::*[self::AND2 or self::OR2])]")
   }
+  if (allow_scoped) {
+    # force 2nd preceding to ensure we're in the loop condition, not the loop expression
+    in_branch_cond <- "ancestor::expr[preceding-sibling::*[2][self::IF or self::WHILE]]"
+    xpath <- paste0(
+      xpath,
+      glue("[not({in_branch_cond}) or expr[1]/SYMBOL = {in_branch_cond}/parent::expr/following::SYMBOL]")
```

Comment:
What does the second or cond do?

Reply:
`expr[1]/SYMBOL = {in_branch_cond}/parent::expr/following::SYMBOL` this one?

It's looking for the symbol on the LHS of `<-` matching a symbol anywhere later in the file. That's very conservative (e.g. it'll match that symbol in NSE and in different environments like other functions), but seems like a reasonable/simple enough start. I'll add a comment.